src/parquet/arrow/schema.cc

```diff
@@ -406,7 +406,7 @@ Status ToParquetSchema(const ::arrow::Schema* arrow_schema,
     RETURN_NOT_OK(FieldToNode(arrow_schema->field(i), properties, &nodes[i]));
   }
 
-  NodePtr schema = GroupNode::Make("schema", Repetition::REPEATED, nodes);
+  NodePtr schema = GroupNode::Make("schema", Repetition::REQUIRED, nodes);
```

Comment:
parquet.thrift seems to think that the root schema should not have any repetition set:

https://github.com/apache/parquet-format/blob/master/src/main/thrift/parquet.thrift#L231

I'm assuming other parquet implementations probably just ignore the root schema's repetition level, but it triggers some logic in Impala to handle nested types.